### ISSUE TITLE
Provide a JSON view alternative to --list-extensions

### ISSUE THREAD

**Issue by aplanas** [SUSE/connect#429]
I need to parse the output of --list-extensions in order to integrate SUSEConnect into SaltStack, but the current implementation of this parameter have two issues:

* The ANSI escape colors are always present, even if I specify a different terminal, or I redirect the output into a file. This makes the output very hard to parse.

* No clear schema definition using only the output as a guide. Some times there are extra elements like activated / deactivated module, requirements for additional registration, etc.

I think that is a good idea to provide a --list-extensions-json parameter to resolve the main issue.

**Comment by lagartoflojo** [comment]
Hi @aplanas thanks for your idea.
We would welcome a PR to add this feature.

**Comment by aplanas** [comment]
Thanks for the quick answer. As today I am faster with Python instead of Ruby, I guess that the sane approach is just forget about this, and there will be no PR from my side.

Thanks again!

**Comment by lagartoflojo** [comment]
Ah, okay.
I would love for our team to be able to help you with this now, but we are completely booked with SUSE separation tasks, and probably will be at least until Apr 2020...

**Comment by aplanas** [comment]
Understood, thanks.
